src/commands/update.rs

```diff
@@ -0,0 +1,484 @@
+        {
+            nsnames.swap(0, mname_idx);
+        }
+
+        Ok(nsnames)
+    }
+
+    /// Create the packet of the update message to send to the name servers
+    fn create_update_message(&self, zone: &Name<Vec<u8>>) -> Vec<u8> {
+        let mut message = MessageBuilder::new_vec();
+
+        let header = message.header_mut();
+        header.set_opcode(Opcode::UPDATE);
+        header.set_qr(false);
+
+        let mut zone_section = message.question();
+        zone_section
+            .push(Question::new(zone, Rtype::SOA, Class::IN))
+            .unwrap();
+
+        let mut update_section = zone_section.authority();
+
+        // If we have an IP address, remove that ip address
+        // else remove A and/or AAAA as defined in RFC2136 2.5.2.
+        if let Some(ip) = self.ip {
+            let rdata: AllRecordData<&[u8], Name<&[u8]>> = match ip {
+                IpAddr::V4(ip) => AllRecordData::A(A::new(ip)),
+                IpAddr::V6(ip) => AllRecordData::Aaaa(Aaaa::new(ip)),
+            };
+            update_section
+                .push(Record::new(
+                    &self.domain,
+                    Class::IN,
+                    Ttl::from_secs(300),
+                    rdata,
+                ))
+                .unwrap();
+        } else {
+            update_section
+                .push(Record::new(
+                    &self.domain,
+                    Class::ANY,
+                    Ttl::from_secs(0),
+                    UnknownRecordData::from_octets(Rtype::A, &[]).unwrap(),
+                ))
+                .unwrap();
+
+            update_section
+                .push(Record::new(
+                    &self.domain,
+                    Class::ANY,
+                    Ttl::from_secs(0),
+                    UnknownRecordData::from_octets(Rtype::AAAA, &[]).unwrap(),
+                ))
+                .unwrap();
+        }
+
+        update_section.finish()
+    }
+
+    /// Send the update packet to the names in nsnames in order until one responds
+    async fn send_update(
+        &self,
+        msg: Vec<u8>,
+        nsnames: Vec<Name<Vec<u8>>>,
+    ) -> Result<(), Error> {
+        let msg = Message::from_octets(msg).unwrap();
+        let resolver = StubResolver::new();
+        for name in nsnames {
+            let found_ips = resolver.lookup_host(&name).await?;
+            for socket in found_ips.port_iter(53) {
+                let response = match &self.tsig {
+                    Some(tsig) => {
+                        let key = Key::new(
+                            tsig.algorithm,
+                            &tsig.key,
+                            tsig.name.clone(),
+                            None,
+                            None,
+                        )
+                        .unwrap();
+                        let msg =
+                            RequestMessageMulti::new(msg.clone()).unwrap();
```

Comment:
What's the reason for using Multi here? I would only expect to see that if XFR responses are expected.

Reply:
It was the only thing that worked with the types...

Reply:
That'd be great! I think there's a missing impl somewhere but forgot to report that back.

Reply:
Done.

Reply:
I'm confused, there's still `RequestMessageMulti` now right?